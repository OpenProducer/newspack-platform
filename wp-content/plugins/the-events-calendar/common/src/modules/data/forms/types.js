/**
 * Internal dependencies
 */
import { PREFIX_COMMON_STORE } from '@moderntribe/common/data/utils';

export const ADD_FORM = `${ PREFIX_COMMON_STORE }/ADD_FORM`;
export const SET_FORM_FIELDS = `${ PREFIX_COMMON_STORE }/SET_FORM_FIELDS`;
export const CREATE_FORM_DRAFT = `${ PREFIX_COMMON_STORE }/CREATE_FORM_DRAFT`;
export const EDIT_FORM_ENTRY = `${ PREFIX_COMMON_STORE }/EDIT_FORM_ENTRY`;
export const SUBMIT_FORM = `${ PREFIX_COMMON_STORE }/SUBMIT_FORM`;
export const CLEAR_FORM = `${ PREFIX_COMMON_STORE }/CLEAR_FORM`;
export const SET_SAVING_FORM = `${ PREFIX_COMMON_STORE }/SET_SAVING_FORM`;

export const ADD_VOLATILE_ID = `${ PREFIX_COMMON_STORE }/ADD_VOLATILE_ID`;
export const REMOVE_VOLATILE_ID = `${ PREFIX_COMMON_STORE }/REMOVE_VOLATILE_ID`;
