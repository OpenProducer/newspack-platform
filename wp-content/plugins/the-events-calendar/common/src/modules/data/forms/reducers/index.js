export { default as form } from './form';
export { default as volatile } from './volatile';
