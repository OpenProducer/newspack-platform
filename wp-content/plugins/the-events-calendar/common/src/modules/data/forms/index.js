/**
 * Internal dependencies
 */
import reducer from './reducer';

import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';

export default reducer;

export { types, actions, selectors };
