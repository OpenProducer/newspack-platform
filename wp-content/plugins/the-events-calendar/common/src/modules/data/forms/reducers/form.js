/**
 * Internal dependencies
 */
import { EVENT } from '@moderntribe/common/data/editor/post-types';
import * as types from '@moderntribe/common/data/forms/types';

export const DEFAULT_STATE = {
	edit: false,
	create: false,
	submit: false,
	saving: false,
	fields: {},
	type: EVENT,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.ADD_FORM:
			return {
				...state,
				type: action.payload.type,
			};
		case types.CLEAR_FORM:
			return {
				...state,
				...DEFAULT_STATE,
				type: state.type,
			};
		case types.CREATE_FORM_DRAFT:
			return {
				...state,
				submit: false,
				edit: false,
				create: true,
				fields: action.payload.fields,
			};
		case types.SET_SAVING_FORM:
			return {
				...state,
				saving: action.payload.saving,
			};
		case types.EDIT_FORM_ENTRY:
			return {
				...state,
				create: false,
				submit: false,
				edit: true,
				fields: action.payload.fields,
			};
		case types.SUBMIT_FORM:
			return {
				...state,
				submit: true,
			};
		default:
			return state;
	}
};
