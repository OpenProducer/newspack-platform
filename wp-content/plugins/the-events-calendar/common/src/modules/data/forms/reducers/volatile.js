/**
 * Internal dependencies
 */
import * as types from '@moderntribe/common/data/forms/types';

export default ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_VOLATILE_ID:
			return [ ...state, action.payload.id ];
		case types.REMOVE_VOLATILE_ID:
			return state.filter( ( id ) => id !== action.payload.id );
		default:
			return state;
	}
};
