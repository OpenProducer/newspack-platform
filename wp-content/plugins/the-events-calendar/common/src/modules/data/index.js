/**
 * Internal Dependencies
 */
import * as editor from './editor/post-types';
import * as forms from './forms';
import * as plugins from './plugins';
export { default } from './reducers';

export { editor, forms, plugins };