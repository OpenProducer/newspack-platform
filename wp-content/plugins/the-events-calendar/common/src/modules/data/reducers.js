/**
 * External dependencies
 */
import { setupCreateReducer } from '@nfen/redux-reducer-injector';
import forms from './forms';

/**
 * Internal dependencies
 */
import plugins from './plugins';

export default setupCreateReducer( {
	plugins,
	forms,
} );
