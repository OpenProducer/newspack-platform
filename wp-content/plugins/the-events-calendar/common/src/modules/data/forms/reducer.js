/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './types';
import { form, volatile } from './reducers';

const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_FORM:
		case types.CLEAR_FORM:
		case types.SET_FORM_FIELDS:
		case types.CREATE_FORM_DRAFT:
		case types.EDIT_FORM_ENTRY:
		case types.SUBMIT_FORM:
		case types.SET_SAVING_FORM:
			return {
				...state,
				[ action.payload.id ]: form( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	volatile,
} );
