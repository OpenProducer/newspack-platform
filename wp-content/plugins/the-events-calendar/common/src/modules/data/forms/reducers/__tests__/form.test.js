/**
 * Internal dependencies
 */
import { form } from '@moderntribe/common/data/forms/reducers';
import { actions } from '@moderntribe/common/data/forms';
import { DEFAULT_STATE } from '@moderntribe/common/data/forms/reducers/form';

describe( '[STORE] - form reducer', () => {
	it( 'Should return the default state', () => {
		expect( form( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should add a new form', () => {
		expect( form( DEFAULT_STATE, actions.registerForm( 20, 'tribe_organizers' ) ) ).toMatchSnapshot();
	} );

	it( 'Should clear a form', () => {
		const state = {
			...DEFAULT_STATE,
			type: 'tribe_organizers',
		};
		expect( form( state, actions.clearForm( 20 ) ) ).toMatchSnapshot();
	} );

	it( 'Should create a form draft', () => {
		const fields = {
			id: 20,
			title: 'Modern Tribe',
		};
		expect( form( DEFAULT_STATE, actions.createDraft( 20, fields ) ) ).toMatchSnapshot();
	} );

	it( 'Should toggle the saving form flag', () => {
		expect( form( DEFAULT_STATE, actions.setSaving( 20, true ) ) ).toMatchSnapshot();
		expect( form( DEFAULT_STATE, actions.setSaving( 20, false ) ) ).toMatchSnapshot();
	} );

	it( 'Should edit the form entry', () => {
		const fields = {
			title: 'Tribe',
			description: '',
		};
		expect( form( DEFAULT_STATE, actions.editEntry( 20, fields ) ) ).toMatchSnapshot();
	} );

	it( 'Should submit the form', () => {
		expect( form( DEFAULT_STATE, actions.setSubmit( 20 ) ) ).toMatchSnapshot();
	} );
} );
