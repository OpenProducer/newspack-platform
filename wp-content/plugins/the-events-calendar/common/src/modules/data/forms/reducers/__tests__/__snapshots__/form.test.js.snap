// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - form reducer Should add a new form 1`] = `
Object {
  "create": false,
  "edit": false,
  "fields": Object {},
  "saving": false,
  "submit": false,
  "type": "tribe_organizers",
}
`;

exports[`[STORE] - form reducer Should clear a form 1`] = `
Object {
  "create": false,
  "edit": false,
  "fields": Object {},
  "saving": false,
  "submit": false,
  "type": "tribe_organizers",
}
`;

exports[`[STORE] - form reducer Should create a form draft 1`] = `
Object {
  "create": true,
  "edit": false,
  "fields": Object {
    "id": 20,
    "title": "Modern Tribe",
  },
  "saving": false,
  "submit": false,
  "type": "tribe_events",
}
`;

exports[`[STORE] - form reducer Should edit the form entry 1`] = `
Object {
  "create": false,
  "edit": true,
  "fields": Object {
    "description": "",
    "title": "Tribe",
  },
  "saving": false,
  "submit": false,
  "type": "tribe_events",
}
`;

exports[`[STORE] - form reducer Should submit the form 1`] = `
Object {
  "create": false,
  "edit": false,
  "fields": Object {},
  "saving": false,
  "submit": true,
  "type": "tribe_events",
}
`;

exports[`[STORE] - form reducer Should toggle the saving form flag 1`] = `
Object {
  "create": false,
  "edit": false,
  "fields": Object {},
  "saving": true,
  "submit": false,
  "type": "tribe_events",
}
`;

exports[`[STORE] - form reducer Should toggle the saving form flag 2`] = `
Object {
  "create": false,
  "edit": false,
  "fields": Object {},
  "saving": false,
  "submit": false,
  "type": "tribe_events",
}
`;
